include/psa/crypto_builtin_composites.h

```diff
@@ -51,7 +51,7 @@ typedef struct {
     uint8_t MBEDTLS_PRIVATE(opad)[PSA_HMAC_MAX_HASH_BLOCK_SIZE];
 } mbedtls_psa_hmac_operation_t;
 
-#define MBEDTLS_PSA_HMAC_OPERATION_INIT { 0, PSA_HASH_OPERATION_INIT, { 0 } }
+#define MBEDTLS_PSA_HMAC_OPERATION_INIT { 0, PSA_HASH_OPERATION_INIT, {} }
```

Comment:
I'm afraid this isn't an acceptable fix: `{}` is not standard C99. It's a GCC extension, and with `gcc -std=c99 -pedantic` this is just trading one warning for another.

The proper fix is to list the correct number of zeros with the correct nesting of braces. This is what we meant to do, but we might have gotten it wrong.

What warning option is triggering? Or what configuration? In the default configuration, the library builds cleanly with `gcc-12 -std=c99 -pedantic -Werror -Wall -Wextra` and `clang-16 -std=c99 -pedantic -Werror -Wall -Wextra`.